src/App.tsx

```diff
@@ -9,51 +9,50 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
 import { Counter } from './features/counter/Counter';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions as postsActions } from './features/postsSlice';
+import { actions as selectedPostActions } from './features/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const { posts, loaded, hasError } = useAppSelector(state => state.posts);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const { author } = useAppSelector(state => state.author);
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
 
   function loadUserPosts(userId: number) {
-    setLoaded(false);
+    dispatch(postsActions.setLoading(false));
 
     getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
+      .then(postsFromServer => {
+        dispatch(postsActions.setPosts(postsFromServer));
+        dispatch(postsActions.setHasError(false));
+      })
+      .catch(() => postsActions.setHasError(true))
+      .finally(() => dispatch(postsActions.setLoading(true)));
```

Comment:
on finally you should set loading false, why true?
```suggestion
      .finally(() => dispatch(postsActions.setLoading(false)));
```